spec/unit/connection_spec.rb

```diff
@@ -14,29 +14,60 @@
 
   describe 'When verifying user groups' do
     before :all do
-      default_devise_settings!
-      reset_ldap_server!
+      #default_devise_settings!
+      #reset_ldap_server!
     end
     context 'through ldap_ad_group_check' do
       it 'should search using user_lookup_attribute and group_lookup_attribute', :focus => true do
-        admin = Factory.create(:admin)
+        @admin = Factory.create(:admin)
+        @group_lookup_attribute = 'memberof'
+        @user_lookup_attribute = 'extensionAttribute2'
+        @group_name = 'cn=admins,ou=groups,dc=test,dc=com'
         ::Devise.ldap_config = Proc.new() {{
             'host' => 'localhost',
             'port' => 3389,
-            'base' => 'ou=testbase,dc=test,dc=com',
-            'group_base' => 'ou=testbase,dc=test,dc=com',
+            'base' => 'dc=test,dc=com',
+            'group_base' => 'ou=groups,dc=test,dc=com',
             'attribute' => 'cn',
-            'user_lookup_attribute' => 'mail',
-            'group_lookup_attribute' => 'memberof',
+            'user_lookup_attribute' => 'extensionAttribute2',
+            'group_lookup_attribute' => 'memberOf',
             'admin_user' => 'cn=admin,dc=test,dc=com',
             'admin_password' => 'secret',
-            'required_groups' => ['testgroup1', 'testgroup2']
+            'required_groups' => ['cn=admins,ou=groups,dc=test,dc=com']
         }}
         ::Devise.ldap_ad_group_check = true
         ::Devise.ldap_check_group_membership = true
-        ::Devise.ldap_auth_username_builder = Proc.new() {|attribute, login, ldap| "#{login}"}
-        connection = Devise::LDAP::Connection.new(:login => admin.email, :password => admin.password, :admin => true)
-        assert_equal true, connection.in_group?('cn=admins,ou=testbase,dc=test,dc=com')
+        ::Devise.ldap_auth_username_builder = Proc.new() {|attribute, login, ldap|
+          "#{login}"
+        }
+
+        def mockResponse(group_lookup_attribute, user_lookup_attribute, group_name)
+          filter = generate_filter(group_lookup_attribute, user_lookup_attribute, group_name)
+          if(filter.to_s.byteslice(1..-1) == "&(#{user_lookup_attribute}=cn=example.admin@test.com,ou=people,dc=test,dc=com)(#{group_lookup_attribute}=cn=admins,ou=groups,dc=test,dc=com))".encode('UTF-8'))
```

Comment:
is this the hack we were talking about? Let's comment on hacks with a TODO: this is a hack for x, needs a better way. type thing